### PR TITLE
Allow custom trace exporter for tsh

### PR DESCRIPTION
**Pull request by rosstimothy**
Trace forwarding via `tsh --trace` only works to date if Auth is configured with the `tracing_service` enabled. In all other scenarios the traces are still forwarded to Auth but are silently dropped. This makes it difficult to capture valuable traces from customers with latency issues as they are first required to setup a Telemetry backend and enable tracing in their cluster.

A new `--trace-exporter` flag is added to `tsh` to make it possible to direct traces from `tsh` to a file or local instance of jaeger without hacing to modify their Teleport cluster. The url must follow the same semantics as the config file equivalent.

One important caveat is that **only** the `tsh` spans will be captured. Any corresponding `teleport` spans are exported acording to the `tracing_service`. While this only paints half the picture, it is still a good indicator of where `tsh` may be experiencing latency.

An example usage to send traces to local files:

```bash
tsh --trace --trace-exporter=file:///some/path/traces ssh user@foo
```

### CODE REVIEW

File: tool/tsh/tsh.go [hunk 1]
```diff
@@ -1161,31 +1174,51 @@ func newTraceProvider(cf *CLIConf, command string, whitelist []string) (*tracing
...
+				SamplingRate: 1.0,
+			})
+		if err != nil {
+			return trace.NewAggregate(err, clt.Close())
+		}
+
+		provider = p
+		return nil
+	}); err != nil {
+		return nil, trace.Wrap(err)
```
Comment (smallinsky) [inline tool/tsh/tsh.go:1221]: nit: For me previous version was more readable where the provider as created outside `client.RetryWithRelogin(` flow.

File: tool/tsh/tsh.go [hunk 2]
```diff
@@ -953,7 +958,12 @@ func Run(ctx context.Context, args []string, opts ...cliOption) error {
 	// Connect to the span exporter and initialize the trace provider only if
 	// the --trace flag was set.
 	if cf.SampleTraces {
-		provider, err := newTraceProvider(&cf, command, []string{login.FullCommand()})
+		// login only needs to be whitelisted if forwarding to auth
+		var whitelist []string
+		if cf.TraceExporter == "" {
```
Comment (smallinsky) [inline tool/tsh/tsh.go:963]: Should we make the `--trace` enabled by default when the `trace-exporter` flag was set ? 
Right now a user needs to provide both flags `--trace --trace-exporter`  but  not handle the `--trace-exporter` with trace flag by default ?

File: lib/observability/tracing/collector.go [hunk 3]
```diff
@@ -45,6 +45,7 @@ type Collector struct {
 
 	spanLock sync.RWMutex
 	spans    []*otlp.ScopeSpans
+	exported chan struct{}
```
Comment (smallinsky) [inline lib/observability/tracing/collector.go:48]:
```suggestion
	exportedC chan struct{}
```

**Review by smallinsky — APPROVED** [review]

File: tool/tsh/tsh.go [hunk 4]
```diff
@@ -365,9 +364,14 @@ type CLIConf struct {
 
 	// ListAll specifies if an ls command should return results from all clusters and proxies.
 	ListAll bool
+
 	// SampleTraces indicates whether traces should be sampled.
 	SampleTraces bool
 
+	// TraceExporter is a manually provided uri provided to send traces to instead of
```
Comment (zmb3) [inline tool/tsh/tsh.go:371]:
```suggestion
	// TraceExporter is a manually provided URI to send traces to instead of
```

File: tool/tsh/tsh.go [hunk 5]
```diff
@@ -954,7 +959,12 @@ func Run(ctx context.Context, args []string, opts ...cliOption) error {
 	// Connect to the span exporter and initialize the trace provider only if
 	// the --trace flag was set.
 	if cf.SampleTraces {
-		provider, err := newTraceProvider(&cf, command, []string{login.FullCommand()})
+		// login only needs to be whitelisted if forwarding to auth
+		var whitelist []string
```
Comment (zmb3) [inline tool/tsh/tsh.go:963]:
```suggestion
		var allowed []string
```

https://www.forbes.com/sites/carolinamilanesi/2021/06/29/the-importance-of-inclusive-language-and-design-in-tech/?sh=ead71d9b1672

**Review by jakule — APPROVED** [review]

**Pull request merged**